geode-serialization/src/main/java/org/apache/geode/internal/serialization/BasicSerializable.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.serialization;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+
+/**
+ * This interface supports toData/fromData serialization. DataSerializableFixedID is almost always
+ * preferable to this interface for new work.
+ */
+public interface BasicSerializable {
+
+  /**
+   * Writes the state of this object as primitive data to the given <code>DataOutput</code>.
+   *
+   * @throws IOException A problem occurs while writing to <code>out</code>
+   */
+  default void toData(DataOutput out, SerializationContext context) throws IOException {};
```

Comment:
I don't like "default" on these methods. It seems better, in this case, to insist that implementors of this interface implement toData and fromData.

Reply:
why does that seem better @dschneider-pivotal ?

Reply:
The defaults make it too easy to implement this interface and have it serialize nothing. By forcing the implementor to implement toData and fromData they get a chance to consider if their class has any data that needs to be serialized/deserialized. I don't think you have any backwards compatibility to be concerned about so you don't need to use default for that reason. And you don't actually have any interesting "default" implementation here (just empty methods).

Reply:
I understand your point. OTOH we (now) clearly document `BasicSerializable` as an interface that is solving a particular legacy problem and we clearly say not to use it for new work. Inasmuch as only 6 classes implement this interface and only 2/6 need to override the (no-op) behavior, I'm gonna leave the `default` impls in.